unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/Tabs/FriendsTabViewBase.cs

```diff
@@ -54,6 +65,46 @@ void UpdateToggle()
             toggleText.text = $"{toggleTextPrefix} ({Count()})";
             toggleButton.SetActive(Count() != 0);
         }
+
+        public void AddOrUpdateLastTimestamp(LastFriendTimestampModel timestamp)
+        {
+            if (timestamp == null)
+                return;
+
+            LastFriendTimestampModel existingTimestamp = latestTimestampsOrdered.FirstOrDefault(t => t.userId == timestamp.userId);
+            if (existingTimestamp == null)
+            {
+                latestTimestampsOrdered.Add(timestamp);
+            }
+            else if (timestamp.lastMessageTimestamp > existingTimestamp.lastMessageTimestamp)
+            {
+                existingTimestamp.lastMessageTimestamp = timestamp.lastMessageTimestamp;
+            }
+
+            latestTimestampsOrdered = latestTimestampsOrdered.OrderByDescending(f => f.lastMessageTimestamp).ToList();
+
+            ReorderingFriendEntries();
+        }
+
+        public LastFriendTimestampModel RemoveLastTimestamp(string userId)
+        {
+            LastFriendTimestampModel timestampToRemove = latestTimestampsOrdered.FirstOrDefault(t => t.userId == userId);
+            if (timestampToRemove == null)
+                return null;
+
+            latestTimestampsOrdered.Remove(timestampToRemove);
+
+            return timestampToRemove;
+        }
+
+        private void ReorderingFriendEntries()
```

Comment:
Is there any way we can avoid reordering the collection on every new message? For example if we get several messages from the same friend, the reordering would be unnecessary after the first message, as that friend would already be on top of the list for the other messages.

Reply:
Good point, let me take a look. Maybe we can control that case and avoid some reordering calls.

Reply:
@pravusjif I have submitted a change for avoid to reordering friend list when the last message came from the same user. This will avoid some innecesary calls, thanks for the review! :)